### PR TITLE
Use login form to confirm access

### PR DESCRIPTION
**Pull request by zero-24**
Use login form to confirm access

### CODE REVIEW

File: administrator/components/com_joomlaupdate/controllers/update.php [hunk 1]
```diff
@@ -358,4 +358,53 @@ protected function _applyCredentials()
...
+	 * Checks the admin has super administrator privileges and then proceeds with the final & cleanup steps.
+	 *
+	 * @return  array
+	 *
+	 * @since   __DEPLOY_VERSION__
+	 */
+	public function confirmfinalise()
+	{
+		// Check for request forgeries
+		JSession::checkToken() or jexit(JText::_('JINVALID_TOKEN'));
```
Comment (andrepereiradasilva) [inline administrator/components/com_joomlaupdate/controllers/update.php]: You cant check the token here. In previous version there is no token at this point. Only after the login submit

File: administrator/components/com_joomlaupdate/views/update/tmpl/finaliseconfirm.php [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+
+defined('_JEXEC') or die;
+
+JHtml::_('behavior.keepalive');
+JHtml::_('bootstrap.tooltip');
+JHtml::_('jquery.framework');
+
+JLoader::register('ModLoginHelper', JPATH_ADMINISTRATOR . '/modules/mod_login/helper.php');
+
+$twofactormethods = ModLoginHelper::getTwoFactorMethods();
```
Comment (andrepereiradasilva) [inline administrator/components/com_joomlaupdate/views/update/tmpl/finaliseconfirm.php:17]: Use JAuthencationHelper só you don't need to load the helper

File: administrator/components/com_joomlaupdate/views/update/tmpl/finaliseconfirm.php [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+					<button tabindex="5" class="btn btn-primary">
+						<span class="icon-play icon-white"></span> <?php echo JText::_('COM_JOOMLAUPDATE_VIEW_UPDATE_FINALISE_CONFIRM_AND_CONTINUE'); ?>
+					</button>
+				</div>
+			</div>
+		</div>
+
+		<input type="hidden" name="option" value="com_joomlaupdate"/>
+		<input type="hidden" name="task" value="update.confirmfinalise" />
+		<input type="hidden" name="method" value="direct" />
```
Comment (andrepereiradasilva) [inline administrator/components/com_joomlaupdate/views/update/tmpl/finaliseconfirm.php]: You don't need this here

File: administrator/components/com_joomlaupdate/views/update/tmpl/finaliseconfirm.php [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+ *
+ * @copyright   Copyright (C) 2005 - 2016 Open Source Matters, Inc. All rights reserved.
+ * @license     GNU General Public License version 2 or later; see LICENSE.txt
+ */
+
+defined('_JEXEC') or die;
+
+JHtml::_('behavior.keepalive');
+JHtml::_('bootstrap.tooltip');
+JHtml::_('jquery.framework');
```
Comment (andrepereiradasilva) [inline administrator/components/com_joomlaupdate/views/update/tmpl/finaliseconfirm.php]: Why do you jquery here?

File: administrator/components/com_joomlaupdate/controllers/update.php (hunk 1 above)
Comment (andrepereiradasilva) [inline administrator/components/com_joomlaupdate/controllers/update.php]: Ups sorry forget

**Comment by zero-24** [comment]
Thanks i can fix it tomorrow.

File: administrator/components/com_joomlaupdate/views/update/tmpl/finaliseconfirm.php (hunk 4 above)
Comment (zero-24) [inline administrator/components/com_joomlaupdate/views/update/tmpl/finaliseconfirm.php]: Just a relict of other testing ;)

File: administrator/components/com_joomlaupdate/views/update/tmpl/finaliseconfirm.php (hunk 3 above)
Comment (zero-24) [inline administrator/components/com_joomlaupdate/views/update/tmpl/finaliseconfirm.php]: Correct i'm going to fix it tomorrow

File: administrator/components/com_joomlaupdate/views/update/tmpl/finaliseconfirm.php (hunk 2 above)
Comment (zero-24) [inline administrator/components/com_joomlaupdate/views/update/tmpl/finaliseconfirm.php:17]: Correct i just need this as i'm patching just the 3.6.2 package and that helper is just part of 3.6.3 but that will be fixed tomorrow too :)

Comment (andrepereiradasilva) [inline administrator/components/com_joomlaupdate/views/update/tmpl/finaliseconfirm.php:17]: Let it be this way if you create a joomlaupdate package it works in ollder version too

File: administrator/components/com_joomlaupdate/controllers/update.php [hunk 5]
```diff
@@ -358,4 +358,53 @@ protected function _applyCredentials()
...
+			'username'  => $this->input->post->get('username', '', 'username'),
+			'password'  => $this->input->post->get('passwd', '', 'raw'),
+			'secretkey' => $this->input->post->get('secretkey', '', 'raw'),
+		);
+
+		$result = $model->captiveLogin($credentials);
+
+		// The login fails?
+		if (!$result)
+		{
```
Comment (andrepereiradasilva) [inline administrator/components/com_joomlaupdate/controllers/update.php:395]: Shouldnt more than one attempt be allowed?

File: administrator/components/com_joomlaupdate/controllers/update.php [hunk 6]
```diff
@@ -358,4 +358,53 @@ protected function _applyCredentials()
...
+			JLog::add(JText::sprintf('COM_JOOMLAUPDATE_UPDATE_LOG_CONFIRM_FINALISE_FAIL'), JLog::INFO, 'Update');
+
+			// Redirect to com_joomlaupdate page
+			$url = 'index.php?option=com_joomlaupdate&' . JFactory::getSession()->getFormToken() . '=1';
+			$this->setRedirect($url);
```
Comment (andrepereiradasilva) [inline administrator/components/com_joomlaupdate/controllers/update.php]: We don't need a token in the main componentes page

**Comment by andrepereiradasilva** [comment]
it's a pain, i only have php 7 servers so i can only test from 3.5.0+

File: administrator/components/com_joomlaupdate/controllers/update.php (hunk 5 above)
Comment (zero-24) [inline administrator/components/com_joomlaupdate/controllers/update.php:395]: hmm I'm not sure if it works but i can try it.

**Comment by zero-24** [comment]
I have just fixed this issues and now i'm going to test the multible login.

**Comment by zero-24** [comment]
@andrepereiradasilva please test it relogin is now possible. And tha package is updated

**Comment by andrepereiradasilva** [comment]
ok will test

**Comment by andrepereiradasilva** [comment]
tested and seems to works fine.

**Comment by zero-24** [comment]
please merge than we can send your branch as pull request.

**Comment by zero-24** [comment]
or should i send my branch?

**Comment by andrepereiradasilva** [comment]
send your please. i'm working on other things. don't have much time fo this one.

**Comment by zero-24** [comment]
ok

**Pull request closed without merging**